version: '3.8'

services:
  pgadmin:
    image: dpage/pgadmin4
    hostname: "{{.Service.Name}}.{{.Task.Slot}}"
    
    deploy:
      replicas: 1
      restart_policy:
        condition: on-failure
      placement:
        constraints: [node.role == manager]
    #depends_on:
    #  - postgres
    volumes:
      - pgadmin-data:/var/lib/pgadmin
    networks:
    - traefik_public  
    #ports:
    #  - 5050:82
    environment:
      - PGADMIN_DEFAULT_EMAIL: marcelu.phd@gmail.com
      - PGADMIN_DEFAULT_PASSWORD: Ma1x1x0x!!
      - PGADMIN_LISTEN_PORT=80
      #SCRIPT_NAME: /_admin/pgadmin
    labels:
      - traefik.enable=true
      - "traefik.swarm.network=traefik_public"

      - traefik.http.routers.pgadmin.entrypoints=web
      - traefik.http.routers.pgadmin.service=pgadmin

      - "traefik.http.routers.pgadmin.rule=Host(`pgadmin.senaia.in`)"
      - "traefik.http.routers.pgadmin.entrypoints=websecure"
      - "traefik.http.routers.pgadmin.tls=true"
      - "traefik.http.routers.pgadmin.tls.certresolver=letsencrypt"

      - "traefik.http.services.pgadmin.loadbalancer.server.port=82"


networks:
  traefik_public:
    external: true